examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -143,15 +143,12 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
 
     Messaging::SendFlags sendFlags;
     System::PacketBufferHandle payloadBuf;
-    char * requestData = nullptr;
+    char requestData[kMsgBufSize];
 
     uint32_t size = gPingArguments.GetEchoReqSize();
-    VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
+    VerifyOrExit(size >= kMsgBufSize, err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
 
-    requestData = static_cast<char *>(chip::Platform::MemoryAlloc(size));
-    VerifyOrExit(requestData != nullptr, err = CHIP_ERROR_NO_MEMORY);
-
-    snprintf(requestData, size, "Echo Message %" PRIu64 "\n", gPingArguments.GetEchoCount());
+    snprintf(requestData, kMsgBufSize, "Echo Message %" PRIu64 "\n", gPingArguments.GetEchoCount());
 
     payloadBuf = MessagePacketBuffer::NewWithData(requestData, size);
```

Comment:
But more to the point, why do we need requestData at all?  Why not just snprintf directly into the packetbuffer?  Because we don't want the trailing null in there?  Something else?

Reply:
NewWithData API is designed to allocates a packet buffer with initial contents.

Reply:
I know what `NewWithData` does.  But per above, we're completely misusing it to start with.  And nothing is stopping us from calling `MessagePacketBuffer::New` and then writing into the buffer, right?  Might make it less likely that we mess up how big we think the `DataLength()` is if we do it that way, and won't need the extra stack buffer.  In particular, we'd just set `DataLength()` to the return value of `snprintf`, right?

Reply:
Ack, change to New